WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
Still... can we unify how we perform this check?

My concern is someone changing this condition, but not the other one because it's written twice inline.  The location of the method could be anywhere we deem appropriate for now.